Frends.AzureEventHub.Receive/Frends.AzureEventHub.Receive/Definitions/Consumer.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace Frends.AzureEventHub.Receive.Definitions;
+
+/// <summary>
+/// Consumer parameters.
+/// </summary>
+public class Consumer
+{
+    /// <summary>
+    /// Authentication method.
+    /// </summary>
+    /// <example>AuthenticationMethod.ConnectionString</example>
+    [DefaultValue(AuthenticationMethod.ConnectionString)]
+    public AuthenticationMethod AuthenticationMethod { get; set; }
+
+    /// <summary>
+    /// The name of the specific Event Hub.
+    /// </summary>
+    /// <example>ExampleHub</example>
+    public string EventHubName { get; set; }
+
+    /// <summary>
+    /// The name of the consumer group. 
+    /// Events are read in the context of this group.
+    /// Using default consumer group if left empty.
+    /// </summary>
+    /// <example>$Default</example>
+    public string ConsumerGroup { get; set; }
+
+    /// <summary>
+    /// The connection string.
+    /// </summary>
+    /// <example>Endpoint=sb://NamespaceName.servicebus.windows.net/;SharedAccessKeyName=KeyName;SharedAccessKey=KeyValue</example>
+    [UIHint(nameof(AuthenticationMethod), "", AuthenticationMethod.ConnectionString)]
+    [PasswordPropertyText]
+    public string ConnectionString { get; set; }
+
+    /// <summary>
+    /// The fully qualified Event Hubs namespace to connect to.
+    /// Required when using SAS Token or OAuth2 authentication method.
+    /// </summary>
+    /// <example>{yournamespace}.servicebus.windows.net</example>
+    public string FullyQualifiedNamespace { get; set; }
+
+    /// <summary>
+    /// The shared access signature. 
+    /// Access controls may be specified by the Event Hubs namespace or the requested Event Hub, depending on Azure configuration.
+    /// </summary>
+    /// <example>sv=2021-04-10&amp;se=2022-04-10T10%3A431Z&amp;sr=c&amp;sp=l&amp;sig=ZJg983RovE%2BZXI</example>
+    [UIHint(nameof(AuthenticationMethod), "", AuthenticationMethod.SASToken)]
+    [PasswordPropertyText]
+    public string SASToken { get; set; }
+
+    /// <summary>
+    /// The Azure Active Directory tenant (directory) Id.
+    /// </summary>
+    /// <example>Y6b1hf2a-80e2-xyz2-qwer3h-3a7c3a8as4b7f</example>
+    [UIHint(nameof(AuthenticationMethod), "", AuthenticationMethod.OAuth2)]
+    [PasswordPropertyText]
+    public string TenantId { get; set; }
+
+    /// <summary>
+    /// The client (application) ID.
+    /// </summary>
+    /// <example>Y6b1hf2a-80e2-xyz2-qwer3h-3a7c3a8as4b7f</example>
+    [UIHint(nameof(AuthenticationMethod), "", AuthenticationMethod.OAuth2)]
+    [PasswordPropertyText]
```

Comment:
Please add annotation for [DisplayFormat(DataFormatString = "Text")] otherwise the input field can default to Expression.